src/main/java/eu/xenit/transformer/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java

```diff
@@ -0,0 +1,55 @@
+package eu.xenit.transformer.arender.alfresco.content.transformer;
+
+import java.io.InputStream;
+import java.util.UUID;
+import org.alfresco.repo.content.transform.ContentTransformerWorker;
+import org.alfresco.service.cmr.repository.ContentReader;
+import org.alfresco.service.cmr.repository.ContentWriter;
+import org.alfresco.service.cmr.repository.TransformationOptions;
+import org.apache.log4j.Logger;
+
+public class ARenderContentTransformerWorker implements ContentTransformerWorker {
+
+    private static final Logger LOGGER = Logger.getLogger(ARenderContentTransformerWorker.class);
+
+    private ContentTransformARenderRestClient restClient;
+
+    @Override
+    public boolean isAvailable() {
+        return restClient.getWeatherPerformance() >= 0;
+    }
+
+    @Override
+    public String getVersionString() {
+        return "1.0.0";
+    }
+
+    @Override
+    public boolean isTransformable(String sourceMimetype, String targetMimetype, TransformationOptions options) {
+        return "application/pdf".equals(targetMimetype);
+    }
+
+    @Override
+    public String getComments(boolean available) {
+        return null;
+    }
+
+    @Override
+    public void transform(ContentReader reader, ContentWriter writer, TransformationOptions options) throws Exception {
+        String uuid = UUID.randomUUID().toString();
+        try (InputStream contentInputStream = reader.getContentInputStream()) {
+            LOGGER.debug("Starting rendition for reader " + reader + " with reference ID " + uuid);
+            restClient.uploadDocument(uuid, contentInputStream, reader.getMimetype());
+            LOGGER.debug("Uploaded document with reference ID " + uuid + " to Arender");
+            try (InputStream inputStream = restClient.getInputStream(uuid, "RENDERED")) {
+                LOGGER.debug("Received rendered document for reference ID " + uuid);
+                writer.putContent(inputStream);
+                LOGGER.debug("Written rendered document for reference ID " + uuid + " to writer " + writer);
+            }
+        }
```

Comment:
Add some catches here in case something goes wrong with the inputStreams?

Reply:
Catch and then rethrow the exception?
Because you need to throw the exception when something goes wrong, else you get stuck with a rendition node without content.

Reply:
Should we keep a list of documents (contenturls) that have a failed rendition? 

If we know that a rendition of that document can crash the rendition service... then maybe better to not try and render it at all ? 

You could do a isAvailable check prior to the rendition and then in the catch clause do a check if isAvailable is still true - > if no, blacklist the document

Reply:
> If we know that a rendition of that document can crash the rendition service... then maybe better to not try and render it at all ?

I don't feel like it's the responsibility of this transformer to do that.

> You could do a isAvailable check prior to the rendition and then in the catch clause do a check if isAvailable is still true - > if no, blacklist the document

Since multiple renditions can happen in parallel, we might not know which exact document crashed arender.
And we also don't have a NodeRef to work with here, so we don't know the document.